### PR TITLE
Add a prerequisites section to API Gateway migration

### PR DESCRIPTION
**Pull request by bszwarc**
**Description**

Changes proposed in this pull request:

- Add prerequisites section - checking if Apis have status

### CODE REVIEW

File: docs/api-gateway/03-04-migration.md [hunk 1]
```diff
@@ -12,7 +12,34 @@ During migration, it may turn out that some resource specifications are too comp
...
+Receiving no results means that you can perform the upgrade. If, however, you can see any Api resources in the output, recreate each Api using this script:
+
+```shell script
+# set variables
+export API_NAME={INSERT_API_NAME_HERE}
+export API_NAMESPACE={INSERT_API_NAMESPACE_HERE}
+# create temporary file
+export TMP_FILE=$(mktemp)
+# save Api to temporary file
+kubectl get api -n ${API_NAMESPACE} ${API_NAME} > ${TMP_FILE}
```
Comment (kubadz) [inline docs/api-gateway/03-04-migration.md:32]:
```suggestion
kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml > ${TMP_FILE}
```

File: docs/api-gateway/03-04-migration.md [hunk 2]
```diff
@@ -12,7 +12,34 @@ During migration, it may turn out that some resource specifications are too comp
...
+# save Api to temporary file
+kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml > ${TMP_FILE}
+# remove Api
+kubectl delete api -n ${API_NAMESPACE} ${API_NAME}
+# remove dependent resources
+kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
+kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
+# recreate Api from saved file
+kubectl apply -f ${TMP_FILE}
+```
```
Comment (kubadz) [inline docs/api-gateway/03-04-migration.md:40]:
```suggestion
```shell script
# set variables
export API_NAME={INSERT_API_NAME_HERE}
export API_NAMESPACE={INSERT_API_NAMESPACE_HERE}
# remove dependent resources
kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
# recreate Api
kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml | kubectl replace --force -f -
```
```

Comment (kubadz) [inline docs/api-gateway/03-04-migration.md:40]:
```suggestion
```shell script
# set variables
export API_NAME={INSERT_API_NAME_HERE}
export API_NAMESPACE={INSERT_API_NAMESPACE_HERE}
# remove dependent resources
kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
# recreate Api
kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml | kubectl replace --force -f -
```
```

File: docs/api-gateway/03-04-migration.md [hunk 3]
```diff
@@ -12,7 +12,34 @@ During migration, it may turn out that some resource specifications are too comp
...
+export TMP_FILE=$(mktemp)
+# save Api to temporary file
+kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml > ${TMP_FILE}
+# remove Api
+kubectl delete api -n ${API_NAMESPACE} ${API_NAME}
+# remove dependent resources
+kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
+kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
+# recreate Api from saved file
+kubectl apply -f ${TMP_FILE}
```
Comment (kubadz) [inline docs/api-gateway/03-04-migration.md:39]:
```suggestion
# set variables
export API_NAME={INSERT_API_NAME_HERE}
export API_NAMESPACE={INSERT_API_NAMESPACE_HERE}
# remove dependent resources
kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
# recreate Api
kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml | kubectl replace --force -f -
```

File: docs/api-gateway/03-04-migration.md [hunk 4]
```diff
@@ -12,7 +12,28 @@ During migration, it may turn out that some resource specifications are too comp
...
-Here are the steps you need to follow to ensure your services are properly migrated:
+## Prerequisites
+
+Before the migration process starts, ensure that all Api resources have a status. To do so, fetch all Apis without the status:
+
+```shell script
+kubectl get apis --all-namespaces -o json | jq '.items | .[] | select(.status == null)'
+```
+
+Receiving no results means that you can perform the upgrade. If you can see any Api resources in the output, recreate each Api using this script:
```
Comment (kazydek) [inline docs/api-gateway/03-04-migration.md:23]:
```suggestion
Receiving no results means that you can perform the upgrade. If you see any Api resources in the output, recreate each Api using this script:
```

File: docs/api-gateway/03-04-migration.md [hunk 5]
```diff
@@ -12,7 +12,28 @@ During migration, it may turn out that some resource specifications are too comp
...
+export API_NAME={INSERT_API_NAME_HERE}
+export API_NAMESPACE={INSERT_API_NAMESPACE_HERE}
+# remove dependent resources
+kubectl delete virtualservice -n ${API_NAMESPACE} ${API_NAME}
+kubectl delete policy -n ${API_NAMESPACE} ${API_NAME} --ignore-not-found
+# recreate Api
+kubectl get api -n ${API_NAMESPACE} ${API_NAME} -o yaml | kubectl replace --force -f 
+```
+
+Once the Apis are recreated check again if the status is present. The output should not include any Api resources, allowing you to proceed with the migration. Follow these steps to ensure your services are properly migrated:
```
Comment (kazydek) [inline docs/api-gateway/03-04-migration.md:36]:
```suggestion
Once the Apis are recreated, check again if the status is present. The output should not include any Api resources, allowing you to proceed with the migration. Follow these steps to ensure your services are properly migrated:
```

Comment (kazydek) [reply docs/api-gateway/03-04-migration.md:36]: services or Services?

Comment (kubadz) [reply docs/api-gateway/03-04-migration.md:36]: services

Comment (bszwarc) [reply docs/api-gateway/03-04-migration.md]: agree :D

**Pull request merged**